src/storage/engine/mod.rs

```diff
@@ -332,6 +332,9 @@ pub struct Cursor<I: Iterator> {
     // the data cursor can be seen will be
     min_key: Option<Vec<u8>>,
     max_key: Option<Vec<u8>>,
+    /// Will be set to `false` when a cursor move hits `min_key` or `max_key`.
+    /// Notice that when iterator itself is invalid, this field is not meant to be `false`.
+    in_bound: bool,
```

Comment:
Makes things more complicated, if we don't use `seek` and `next`'s return type but use `valid` to indicate if iterator is valid, there will be no such problem.

Reply:
We still need this flag, since `seek` & `next`'s return type is correct before applying this PR. This PR is to fix the incorrect return value from `valid()`.